### PR TITLE
fixes #3566 - exposes orchestration tasks via the API(v2).

### PR DESCRIPTION
**Pull request by ohadlevy**
this patch create a new route:

/api/orchestration/id/tasks

### CODE REVIEW

**Comment by domcleal** [comment]
Where does the task ID come from?

**Comment by ohadlevy** [comment]
it comes from a field called progress_report_id, in the UI its a hidden field --> https://github.com/theforeman/foreman/blob/develop/app/views/hosts/_form.html.erb#L76

I'm adding support in foreman-background, to respond with the task id / url for async api host creation, but for now, in order to use it, the consumer would need to specify the uuid for himself (as he wont get any reply if its sync).

**Comment by domcleal** [comment]
It seems to me this is of limited use in Foreman core, if there's no way to use it with the rest of the API?

**Comment by ohadlevy** [comment]
Sure you can, when you create a new host, you just need to pass a progress_report_id => UUID, and while foreman is working, you go to 

```
/orchestration/UUID/tasks
```

**Comment by domcleal** [comment]
Ah, gotcha.  Could you add that parameter to the host API docs please?

**Comment by ohadlevy** [comment]
updated

File: config/routes/api/v2.rb [hunk 1]
```diff
@@ -264,6 +264,7 @@
 
         end
       end
+      get 'orchestration/(:id)/tasks', to: 'tasks#index'
```
Comment (domcleal) [inline config/routes/api/v2.rb]: This is the Ruby 1.9 hash syntax, it's breaking 1.8 tests.

Comment (ohadlevy) [inline config/routes/api/v2.rb]: updated, thanks, not sure i how i missed that ;)

**Comment by domcleal** [comment]
More test failures:

Expected [] to not be equal to []. (test_0486_route api/v2/tasks/index should have a permission that grants access)

**Comment by ohadlevy** [comment]
updated

**Comment by ohadlevy** [comment]
[test]

**Comment by domcleal** [comment]
Thanks @ohadlevy, merged as bb3916d62 for Foreman 1.4.0.

**Pull request closed without merging**